test/integration-tests/Seaport.test.ts

```diff
@@ -141,4 +157,187 @@ describe('Seaport', () => {
+      const id = criteriaResolvers[0].identifier
+      // transfer nft to alice as tubbyCatOwner
+      const prevTubbyCatOwner = await tubbyCats.connect(alice).ownerOf(id)
+      await tubbyCats
+        .connect(await ethers.getImpersonatedSigner(prevTubbyCatOwner))
+        .transferFrom(prevTubbyCatOwner, alice.address, id)
+
+      const wethReceived = BigNumber.from(advancedOrder.parameters.offer[0].startAmount).sub(value)
+
+      tubbyCats = tubbyCats.connect(alice)
+      const ownerBefore = await tubbyCats.ownerOf(id)
+      expect(ownerBefore).to.eq(alice.address)
+
+      const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        criteriaResolvers,
+        OPENSEA_CONDUIT_KEY,
+        ADDRESS_ZERO, // 0 addr so router custody
+      ])
+
+      // TODO: need to add arg for msg.value?
+      planner.addCommand(CommandType.SEAPORT_SELL_721, [
+        calldata,
+        tubbyCats.address,
+        OPENSEA_CONDUIT,
+        id,
+        alice.address,
+      ])
+      planner.addCommand(CommandType.SWEEP, [WETH.address, MSG_SENDER, 0])
+      const { commands, inputs } = planner
+
+      const wethBefore = await weth.connect(alice).balanceOf(alice.address)
+      expect(wethBefore).to.eq(0)
+      // put NFT in the router TODO replace with Permit2 721
+      await tubbyCats.transferFrom(alice.address, router.address, id)
+      expect(await tubbyCats.ownerOf(id)).to.eq(router.address)
+
+      const ethBefore = await ethers.provider.getBalance(alice.address)
+      // Send no value here bc not sending ETH
+      const receipt = await (await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, {})).wait()
+
+      const ownerAfter = await tubbyCats.ownerOf(id)
+      const wethAfter = await weth.connect(alice).balanceOf(alice.address)
+      const ethAfter = await ethers.provider.getBalance(alice.address)
+      const gasSpent = getTxGasSpent(receipt)
+      const ethDelta = ethBefore.sub(ethAfter)
+
+      expect(ownerBefore).to.eq(alice.address)
+      expect(ownerAfter.toLowerCase()).to.eq(params.offerer.toLowerCase())
+      expect(wethAfter.sub(wethBefore)).to.eq(wethReceived)
+      expect(ethDelta).to.eq(gasSpent)
+    })
+
+    it('revertable order returns NFT to user if reverts', async () => {
+      let { advancedOrder, criteriaResolvers } = getAdvancedOrderParams(seaportOrders[2])
+      const id = criteriaResolvers[0].identifier
+      // transfer nft to alice as tubbyCatOwner
+      const prevTubbyCatOwner = await tubbyCats.connect(alice).ownerOf(id)
+      await tubbyCats
+        .connect(await ethers.getImpersonatedSigner(prevTubbyCatOwner))
+        .transferFrom(prevTubbyCatOwner, alice.address, id)
+      tubbyCats = tubbyCats.connect(alice)
+      const ownerBefore = await tubbyCats.ownerOf(id)
+      expect(ownerBefore).to.eq(alice.address)
+
+      // Signature is invalid
+      advancedOrder.signature = '0xdeadbeef'
```

Comment:
nit: neater to do this when defining advanced order, so its instantiation is happening all in one place. might do
```
let { advancedOrder: invalidOrder, criteriaResolvers } = getAdvancedOrderParams(seaportOrders[2])
invalidOrder.signature = '0xdeadbeef'
```

Reply:
Great suggestion ty!